Rdmp.Dicom.Tests/Unit/PacsFetch.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            public void OnNotify(object sender, NotifyEventArgs args)
+            {
+                if (args.ProgressEventType==ProgressEventType.Error)
+                    throw new ApplicationException(args.Message);
+            }
+
+            public void OnProgress(object sender, ProgressEventArgs args)
+            {
+            }
+        }
+
+        class QRService : DicomService, IDicomServiceProvider, IDicomCFindProvider, IDicomCEchoProvider,
+            IDicomCMoveProvider
+        {
+            public QRService(INetworkStream stream, Encoding fallbackEncoding, Logger log) : base(stream, fallbackEncoding, log)
+            {
+            }
+
+            public DicomCEchoResponse OnCEchoRequest(DicomCEchoRequest request)
+            {
+                throw new NotImplementedException();
+            }
+
+            public IEnumerable<DicomCFindResponse> OnCFindRequest(DicomCFindRequest request)
+            {
+                throw new NotImplementedException();
+            }
+
+            public IEnumerable<DicomCMoveResponse> OnCMoveRequest(DicomCMoveRequest request)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnConnectionClosed(Exception exception)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnReceiveAbort(DicomAbortSource source, DicomAbortReason reason)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task OnReceiveAssociationReleaseRequestAsync() => SendAssociationReleaseResponseAsync();
+
+            public Task OnReceiveAssociationRequestAsync(DicomAssociation association)
+            {
+                foreach (var pc in association.PresentationContexts)
+                {
+                    if (pc.AbstractSyntax == DicomUID.Verification
+                    || pc.AbstractSyntax==DicomUID.PatientRootQueryRetrieveInformationModelFIND
+                    || pc.AbstractSyntax==DicomUID.PatientRootQueryRetrieveInformationModelMOVE
+                    || pc.AbstractSyntax==DicomUID.StudyRootQueryRetrieveInformationModelFIND
+                    || pc.AbstractSyntax==DicomUID.StudyRootQueryRetrieveInformationModelMOVE)
+                    {
+                        pc.AcceptTransferSyntaxes(DicomTransferSyntax.ExplicitVRLittleEndian);
+                    } else if (pc.AbstractSyntax.StorageCategory != DicomStorageCategory.None)
+                    {
+                        pc.AcceptTransferSyntaxes();
+                    }
+                    else
+                    {
+                        pc.SetResult(DicomPresentationContextResult.RejectAbstractSyntaxNotSupported);
+                    }
+                }
+                return SendAssociationAcceptAsync(association);
+            }
+        }
+
+        private static readonly NopProgressListener Listener = new NopProgressListener();
```

Comment:
You can use `ThrowImmediatelyDataLoadEventListener` instead of having to have a private class.